packages/core/src/modules/protection.ts

```diff
@@ -2,6 +2,39 @@ import _ from "lodash";
 import { Context } from "../context";
 import { getSheetByIndex } from "../utils";
 
+export function checkProtectionCellCanEdit(
+  ctx: Context,
+  r: number,
+  c: number,
+  sheetId: string
+) {
+  const sheetFile = getSheetByIndex(ctx, sheetId);
+  if (_.isNil(sheetFile)) {
+    return true;
+  }
+
+  if (_.isNil(sheetFile.config) || _.isNil(sheetFile.config.authority)) {
+    return true;
+  }
+
+  const aut = sheetFile.config.authority;
+
+  if (_.isNil(aut) || _.isNil(aut.sheet) || aut.sheet === 0) {
```

Comment:
I think that the `lo` field should not be restricted to just authority sheets and should be kept as a generic so that cells in non-authority sheets can be made non-editable too. This will resolve #368.

Essentially, if the sheet is authority sheet then the code submitted by this PR is correct. But if it is not an authority sheet, then check the value of `lo`, if it is `1` then return `true` else return `false`. This will help others leverage your change with the `lo` field as well in a generic manner.

Reply:
Basically replace the `checkProtectionCellCanEdit` function with this - 

```
export function checkProtectionCellCanEdit(
  ctx: Context,
  r: number,
  c: number,
  sheetId: string
) {
  const sheetFile = getSheetByIndex(ctx, sheetId);
  if (_.isNil(sheetFile)) {
    return true;
  }

  if (_.isNil(sheetFile.config) || _.isNil(sheetFile.config.authority)) {
    return true;
  }

  const aut = sheetFile.config.authority;

  const isNotAuthority = _.isNil(aut) || _.isNil(aut.sheet) || aut.sheet === 0;

  const { data } = sheetFile;
  const cell = data?.[r]?.[c];

  if (isNotAuthority) {
    if (cell?.lo === 1) return false;
    return true;
  }
  // otherwise sheet is authority sheet
  if (cell?.lo === 0) return true;
  return false;
}
```